# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1552938493 0
#      Mon Mar 18 19:48:13 2019 +0000
# Node ID cfa2a65c9fdc33ced1a1440756e2f9d46b102ca1
# Parent  72fab319b47d19c4e5cf86531dd977b4f7187802
fix sunlinsol_klu.h detection with sundials 4.x (bug #55950)

* m4/acinclude.m4 (OCTAVE_CHECK_SUNDIALS_SUNLINSOL_KLU): Check for
sunlinsol_klu.h outside of OCTAVE_CHECK_LIB macro.

diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -2234,18 +2234,27 @@ AC_DEFUN([OCTAVE_CHECK_SUNDIALS_SIZEOF_R
     OCTAVE_CONFIGURE_WARNING([warn_sundials_realtype])
   fi
 ])
 dnl
 dnl Check whether SUNDIALS IDA library is configured with SUNLINSOL_KLU
 dnl enabled.
 dnl
 AC_DEFUN([OCTAVE_CHECK_SUNDIALS_SUNLINSOL_KLU], [
+  ## Including <sunlinsol/sunlinsol_klu.h> may depend on including
+  ## <suitesparse/klu.h> first.  So perform the check as follows instead
+  ## of using OCTAVE_CHECK_LIB to check for sunlinsol_klu.h.
+  AC_CHECK_HEADERS([suitesparse/klu.h])
+  AC_CHECK_HEADERS([sunlinsol/sunlinsol_klu.h], [], [],
+    [#ifdef HAVE_SUITESPARSE_KLU_H
+     # include <suitesparse/klu.h>
+     #endif
+    ])
   OCTAVE_CHECK_LIB(sundials_sunlinsolklu, SUNLINSOL_KLU, [],
-    [sunlinsol/sunlinsol_klu.h], [SUNKLU], [],
+    [], [SUNKLU], [],
     [don't use SUNLINSOL_KLU library],
     [AC_CACHE_CHECK([whether compiling a program that calls SUNKLU works],
       [octave_cv_sundials_sunlinsol_klu],
       [AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
          #if defined (HAVE_IDA_IDA_H)
          #include <ida/ida.h>
          #endif
          #if defined (HAVE_KLU_H)
